File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/01-shortestWordDistance.java
class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        /**
        * T: O(n) and S: O(1)
        */
        int iw1 = -1;
        int iw2 = -1;
        int minDist = words.length;
        
        for (int i = 0; i < words.length; i++) {
            if (words[i].equals(word1)) {
                iw1 = i;
            } else if (words[i].equals(word2)) {
                iw2 = i;
            }
            if (iw1 != -1 && iw2 != -1)
                minDist = Math.min(minDist, Math.abs(iw2 - iw1));
        }
        return minDist;
    }
}

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/01-shortestWordDistance.py
class Solution:
    def shortestDistance(self, words: List[str], word1: str, word2: str) -> int:
        '''
        T: O(n) and S: O(1)
        '''
        minDist = len(words)
        iw1, iw2 = None, None
        
        for i, word in enumerate(words):
            if word == word1:
                iw1 = i
            elif word == word2:
                iw2 = i
                
            if iw1 != None and iw2 != None:
                minDist = min(minDist, abs(iw2 - iw1))
            
        return minDist
